tests/src/checkpoint_light_client/mod.rs

```diff
@@ -382,11 +388,46 @@ async fn replay_back_and_updating() -> Result<()> {
     let gas_limit = calculate_gas::<replay_back_io::Process>(&api, program_id, &headers).await?;
     println!("replay_back_io::Process gas_limit = {gas_limit}");
     let result = service
-        .process(headers)
+        .process(headers.clone())
         .send_recv(program_id)
         .await
         .unwrap();
 
+    listener
+        .proc_many(
+            |event| match event {
+                gclient::Event::Gear(gclient::GearEvent::UserMessageSent {
+                    message,
+                    expiration: _,
+                }) => {
+                    if message.source.0 == program_id.into_bytes()
+                        && message.destination.0 == [0; 32]
+                    {
+                        let event =
+                            ServiceReplayBackEvents::decode_event(&message.payload.0).unwrap();
+                        match event {
+                            ServiceReplayBackEvents::NewCheckpoint {
+                                slot,
+                                tree_hash_root,
+                            } => {
+                                println!("event = {event:?}");
+                                assert!(headers.iter().any(|header| {
+                                    header.slot == slot && header.tree_hash_root() == tree_hash_root
+                                }));
+                            }
+                        }
```

Comment:
Since the enum contains only one item I guess we may use 
```suggestion
                                println!("event = {event:?}");
let ServiceReplayBackEvents::NewCheckpoint {
                                slot,
                                tree_hash_root,
                            } = event;
                                assert!(headers.iter().any(|header| {
                                    header.slot == slot && header.tree_hash_root() == tree_hash_root
                                }));
```